### PR TITLE
Site name tracks events continued...

### PR DESCRIPTION
**Pull request by mkevins**
This is a continuation of https://github.com/wordpress-mobile/WordPress-Android/pull/16330. In order to work asynchronously, this is based on that PR and can be merged directly to that PR. cc: @ovitrif 

This PR adds tests for the tracking events, and completes the tracking implementation described in the "parent" PR.

To test:

Follow testing instructions on the base PR.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModel.kt [hunk 1]
```diff
@@ -38,26 +38,23 @@ class SiteCreationSiteNameViewModel @Inject constructor(
...
+        analyticsTracker.trackSiteNameCanceled()
         _onBackButtonPressed.call()
     }
 
     fun onSiteNameEntered() {
-        // TODO: analyticsTracker.trackSiteNameEntered(siteName?) maybe we don't want to track the names here?
         uiState.value?.siteName.let {
+            analyticsTracker.trackSiteNameEntered(it)
             if (it.isNullOrBlank()) return
             Log.d("siteNameWip", "Site name entered: $it")
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModel.kt]: I forgot to remove this :man_facepalming:

File: WordPress/src/test/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.wordpress.android.ui.sitecreation.misc.SiteCreationTracker
+import org.wordpress.android.util.LocaleManagerWrapper
+
+@RunWith(MockitoJUnitRunner::class)
+class SiteCreationSiteNameViewModelTest {
+    @Rule
+    @JvmField val rule = InstantTaskExecutorRule()
+
+    @Mock lateinit var analyticsTracker: SiteCreationTracker
+    @Mock lateinit var localeManagerWrapper: LocaleManagerWrapper
```
Comment (mkevins) [inline WordPress/src/test/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModelTest.kt]: I guess this isn't needed here.

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModel.kt (hunk 1 above)
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModel.kt]: Cleaned up with https://github.com/wordpress-mobile/WordPress-Android/pull/16335/commits/b2379edd2c316246046c6f49109d21f5de54083f.

File: WordPress/src/test/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModelTest.kt (hunk 2 above)
Comment (mkevins) [reply WordPress/src/test/java/org/wordpress/android/ui/sitecreation/sitename/SiteCreationSiteNameViewModelTest.kt]: Cleaned up with https://github.com/wordpress-mobile/WordPress-Android/pull/16335/commits/b2379edd2c316246046c6f49109d21f5de54083f.

**Review by ovitrif — APPROVED** [review]
Awesome work @mkevins , thanks a lot for proceeding forward with this 🙏 

All good from my side, I'm going to merge this into the parent branch, & ~I'll look into adjusting or removing the `Log.d` entries while also~ test everything and preparing that PR for review 🎉 

Thank you for also fixing the parts I didn't completely got right in my draft 🙇 

**Edit:** Nvm the strike-through text, the `Log.d` lines were removed here 👍

**Pull request merged**